bagvalidate/validator.go

```diff
@@ -10,8 +18,35 @@ func (v noopValidator) Validate(path string) error {
 	return nil
 }
 
+type validator struct{}
```

Comment:
I think this is backwards, "Validator" should be the interface (more general) whereas "bagValidator" should be a struct implementing the `validator` interface (more specific).